src/configuration/applier/macros.cc

```diff
@@ -164,8 +145,8 @@ void applier::macros::_set_macro(
        std::string const& value) {
   if (type >= MACRO_X_COUNT)
     throw (engine_error() << "Invalid type of global macro: " << type);
-  if (!_mac->x[type] || strcmp(_mac->x[type], value.c_str()))
-    string::setstr(_mac->x[type], value);
+  if (_mac->x[type] != value.c_str())
```

Comment:
I supppose this is a comparison between two strings. So why using c_str().

Reply:
my bad, I missed it.